### ISSUE TITLE
Modbus queue filling up

### ISSUE THREAD

**Issue by pnomme** [GMTrevis/Homeassistant-NodeRed-Systemair-VTR300#6]
Hei Glenn Martin, Great work :) 
I´ve got it up and running with my VSR500 and it is excellent..with one exception;

The queue keeps filling up, resulting in unstable performance and a persistent notification every few minutes: 
<img width="632" alt="image" src="https://user-images.githubusercontent.com/19797696/174426969-2ada3933-6156-4c84-bfb0-3795b01760a2.png">

I am using Modbus-TCP with the IAM adapter and a Wi-Fi connection. The VSR500 has the same Modbus settings as the VTR300, at least I dont get any errors on that part.

Any suggestions?

mvh Peter

**Comment by GMTrevis** [comment]
Hi,
Thanks:)

As far as i know the hardware is the same for both 300, 500 and other Systemair Save Units.
Im far from a network expert but i do believe that the Wifi network (lost packets) could be causing this.
I'm also using the IAM module but the connection is hardwired (not Wifi).
Earlier i also had this issue, my conclusion was that i was polling too many modbus registers too frequent (the unit did not manage respond quickly enough to the requested messages), reducing the poll time eliminated the queue build up. You have the same poll speed that works for me as the code was shared after the changing the poll speed/time.

I think your Modbus TCP over wifi connection can be causing dropped/missed response messages from the unit which causes the queue build up. Try stop polling modbus registers for the schedule and see if that reduce the queue build up (Disable the Inject Nodes "Poll (100s)", "Poll (105s)" and more if necessary.
Maybe your Wifi network performance could be causing this (just thinking loud here).

I hope this helps to solve the issues you are facing:)

Br
GMT

**Comment by pnomme** [comment]
Yup, going for a wired connection did the trick :) Actually, I am now using a gigabit Powerline adapter instead of Wi-Fi, and the queue now is fluxing from 0 - 30ish range and everything seems to be responding as supposed to. Apparently, the IAM hw/cpu is too weak to handle the traffic..

Problem solved :)

Thanks,
Peter
